src/core/Windows.zig

```diff
@@ -205,18 +203,28 @@ fn wndProc(wnd: w.HWND, msg: u32, wParam: w.WPARAM, lParam: w.LPARAM) callconv(w
             const width: u32 = @as(u32, @intCast(lParam & 0xFFFF));
             const height: u32 = @as(u32, @intCast((lParam >> 16) & 0xFFFF));
 
-            window.width = width;
-            window.height = height;
+            if (core_window.width != width or core_window.height != height) {
+                // Recreate the swap_chain
+                core_window.swap_chain.release();
+                core_window.swap_chain_descriptor.width = width;
+                core_window.swap_chain_descriptor.height = height;
+                core_window.swap_chain = core_window.device.createSwapChain(core_window.surface, &core_window.swap_chain_descriptor);
 
-            core.pushEvent(.{ .window_resize = .{ .window_id = window_id, .size = .{ .width = width, .height = height } } });
+                core_window.width = width;
+                core_window.height = height;
+                core_window.framebuffer_width = width;
+                core_window.framebuffer_height = height;
```

Comment:
This is wrong / a bug - the window width/height should almost certainly not be the same as framebuffer_width/framebuffer_height.